src/backend/config.py

```diff
@@ -54,17 +57,31 @@ def get_api_path(ui_path: str) -> str:
     return f"{ui_path.rstrip('/')}/api"
 
 
+def get_s3_addressing_style() -> S3AddressingStyle:
+    """
+    Obtain the S3 addressing style. Defaults to `auto`.
+    """
+    addressing_style: str | None = os.environ.get("S3GW_S3_ADDRESSING_STYLE")
+    if addressing_style is not None:
+        addressing_style = addressing_style.lower()
+    if addressing_style not in ["auto", "virtual", "path"]:
+        addressing_style = "auto"
+    logger.info(f"Using '{addressing_style}' S3 addressing style")
+    return addressing_style  # noqa # pyright: ignore [reportGeneralTypeIssues]
```

Comment:
if the default is supposed to be `auto`, I think that can be passed to `os.environ.get()` as the default. And if it doesn't match anything on the list, I think we should be throwing an error instead of silently converting it to `auto`.

Reply:
Also, not entirely convinced the general type issue there can't be overcome. Maybe declaring `addressing_style` as a `S3AddressingStyle` will address that, assuming you have `auto` as the default as I described above (or as a `S3AddressingStyle | None` if not).

Reply:
Switching to enum makes this code much more simple.